### PR TITLE
Change array type to :array, handle null arrays and json

### PR DESCRIPTION
**Pull request by sreephg**
Snowpack was barfing when Snowflake returns a value of `NULL` in a column of type `ARRAY` (which we map to `:json` in snowpack). 

1. Handled null case for parsing column of type :json
    - This is the only required change, but 2 & 3 will make life easier in the platform
2. Changed ARRAY to map to type `:array` rather than json, so we can return [] rather than :null.
3. Handled null case for :array

### CODE REVIEW

File: lib/snowpack/type_parser.ex [hunk 1]
```diff
@@ -42,6 +42,10 @@ defmodule Snowpack.TypeParser do
 
   defp parse({:integer, data}) when is_binary(data), do: String.to_integer(data)
 
+  defp parse({:array, :null}), do: []
+  defp parse({:array, data}), do: Jason.decode!(data)
+
+  defp parse({:json, :null}), do: :null
```
Comment (fcarlislehg) [inline lib/snowpack/type_parser.ex:48]: Should this return something that is parseable as json like `{}`?

Comment (fcarlislehg) [reply lib/snowpack/type_parser.ex:48]: Similar to how you return an empty array.

Comment (sreephg) [reply lib/snowpack/type_parser.ex:48]: Fair point. That will probably be what any uses of `OBJECT` would wanna use

Comment (sreephg) [reply lib/snowpack/type_parser.ex]: Changed to `{}`

File: lib/snowpack/type_parser.ex [hunk 2]
```diff
@@ -42,6 +42,10 @@ defmodule Snowpack.TypeParser do
 
   defp parse({:integer, data}) when is_binary(data), do: String.to_integer(data)
 
+  defp parse({:array, :null}), do: []
+  defp parse({:array, data}), do: Jason.decode!(data)
+
+  defp parse({:json, :null}), do: {}
```
Comment (fcarlislehg) [inline lib/snowpack/type_parser.ex:48]: I think `Jason.decode` returns a `Map`, so you will want `%{}`.

Comment (fcarlislehg) [reply lib/snowpack/type_parser.ex:48]: `{}` is a `Tuple`

**Review by fcarlislehg — APPROVED** [review]

**Comment by fcarlislehg** [comment]
@sreephg I rebased your commits and used the Semantic Release [commit message format](https://semantic-release.gitbook.io/semantic-release/#commit-message-format) so that the release version would be created correctly.

I also updated to use our new shared GHA for Semantic Release and common configs. This fixed the CI errors.

**Pull request merged**